db/db.pl

```diff
@@ -2461,6 +2497,147 @@ sub dataNodes
         usersCreate();
         queriesCreate();
     }
+} elsif ($ARGV[1] =~ /^rollback$/) {
+
+    logmsg "It is STRONGLY recommended that you stop ALL moloch captures and viewers before proceeding.\n";
+
+    waitFor("ROLLBACK", "do you want to rollback?");
+
+    dbCheckForActivity();
+
+    logmsg "Starting Rollback...\n";
+
+    logmsg "Erasing all indexes except Sessions and History...\n";
+    esDelete("/${PREFIX}tags_v3", 1);
+    esDelete("/${PREFIX}tags_v2", 1);
+    esDelete("/${PREFIX}tags", 1);
+    esDelete("/${PREFIX}sequence", 1);
+    esDelete("/${PREFIX}sequence_v1", 1);
+    esDelete("/${PREFIX}sequence_v2", 1);
+    esDelete("/${PREFIX}files_v5", 1);
+    esDelete("/${PREFIX}files_v4", 1);
+    esDelete("/${PREFIX}files_v3", 1);
+    esDelete("/${PREFIX}files", 1);
+    esDelete("/${PREFIX}stats", 1);
+    esDelete("/${PREFIX}stats_v1", 1);
+    esDelete("/${PREFIX}stats_v2", 1);
+    esDelete("/${PREFIX}stats_v3", 1);
+    esDelete("/${PREFIX}dstats", 1);
+    esDelete("/${PREFIX}dstats_v1", 1);
+    esDelete("/${PREFIX}dstats_v2", 1);
+    esDelete("/${PREFIX}dstats_v3", 1);
+    esDelete("/${PREFIX}fields", 1);
+    esDelete("/${PREFIX}fields_v1", 1);
+    esDelete("/${PREFIX}fields_v2", 1);
+    esDelete("/${PREFIX}hunts_v1", 1);
+    esDelete("/${PREFIX}hunts", 1);
+    esDelete("/${PREFIX}users_v3", 1);
+    esDelete("/${PREFIX}users_v4", 1);
+    esDelete("/${PREFIX}users_v5", 1);
+    esDelete("/${PREFIX}users_v6", 1);
+    esDelete("/${PREFIX}users", 1);
+    esDelete("/${PREFIX}queries", 1);
+    esDelete("/${PREFIX}queries_v1", 1);
+    esDelete("/${PREFIX}queries_v2", 1);
+    esDelete("/_template/${PREFIX}template_1", 1);
+    esDelete("/_template/${PREFIX}sessions_template", 1);
+    esDelete("/_template/${PREFIX}sessions2_template", 1);
+    esDelete("/_template/${PREFIX}history_v1_template", 1);
+
+    my @indexes = ("users", "sequence", "stats", "queries", "hunts", "files", "fields", "dstats");
+    logmsg "Importings settings...\n";
+    foreach my $index (@indexes) { # import settings
+        if (-e "$ARGV[2].$index.settings.json") {
+            open(my $fh, "<", "$ARGV[2].$index.settings.json") or die "cannot open > $ARGV[2].$index.settings.json: $!";
```

Comment:
Thoughts about removing the die and change to warn or something?

Reply:
Removed the “die” statement as there is defensive codes to check a file is found in the directory before to open it.